### PR TITLE
Enable SVG image assets

### LINKED ISSUE #1088: Implement .svg support in Expensify.cash
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/master/CONTRIBUTING.md) for onboarding!
___

**Problem**
We currently use .png or .jpg images in Expensify.cash for things like logos and icons. With these file formats, we need to export the images at multiple sizes for various screen densities like @2x and @3x. For assets that have multiple states like active vs. inactive or like when a conversation is pinned vs. unpinned, we need to provide assets for each state assuming that the different states use different colors. 

**Solution**
Let's implement support for scalable vector graphics (`.svg`) images. The main benefits we receive with `.svg` images are:
- we only need to provide one asset and it will scale to any size in crisp, high resolution
- we can recolor the same asset for multiple states by using the `fill` property
- image sizes will be smaller resulting in a smaller app size

To implement the `.svg` solution, we're recommending the use of the `react-native-svg` library found here: https://github.com/react-native-svg/react-native-svg

For this initial proof of concept to be considered successful, we'd like to try implementing an `.svg` icon for the "pinned" state of a conversation. When a chat is pinned, it will always remain visible in the Left Hand Navigation even after you navigate away from it. There is a small pin icon in the top right header area of a chat to indicate if a conversation is pinned or not:

Pinned | Unpinned
----- | -----
image | image

The `.svg` icon you can use can be found in this .zip file here: [pin.svg.zip](https://github.com/Expensify/Expensify/files/5737657/pin.svg.zip)

The pin icon asset looks like this:
image

While the asset is provided in black, we actually want it to display in our brand colors depending on the state:
- Enabled: `themeColors.heading`
- Disabled: `themeColors.icon`

You can access `themeColor` variables by importing `themeColors from '/src/styles/themes/default.js'`

### PR DESCRIPTION
**Pull request by roryabraham**
cc @shawnborton

### Details
This enables the use of SVG assets in the Expensify.cash codebase.

One can pass a variable fill color to a SVG asset as is done in this PR, BUT:

1) It will only work with one-color assets
2) The assets themselves must not specify a fill color in the internals of the `svg`

This is because the fill passed in via JS is passed to the parent `svg` element, not individual `polygons`, `circles`, `paths`, etc...

We can work around this and gain finer control when necessary, by following this tutorial: https://blog.usejournal.com/how-to-use-svg-in-react-native-2-4140b146b7d1

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/149127
Fixes https://github.com/Expensify/Expensify.cash/issues/1088

### Tests
1) Run the app and sign in
1) Open an unpinned chat
1) Verify that the pin icon is visible and a dull gray
1) Tap or click the pin icon, and watch it become a charcoal
1) Tap the icon again, and watch it return to the dull gray it was before

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

##### Disabled
image

##### Enabled
image

##### Disabled
image

##### Enabled
image

#### iOS
https://user-images.githubusercontent.com/47436092/103324206-48af5080-49fb-11eb-9206-c0def9cbd70c.mov

##### Disabled
image

##### Enabled
image

### CODE REVIEW

**Comment by shawnborton** [comment]
Wow, super cool @roryabraham ! For completeness, should we also try to implement a full color svg, perhaps for the top left icon logo?

**Comment by shawnborton** [comment]
Here is the app icon if you want to try it: [Expensify.cash.svg.zip](https://github.com/Expensify/Expensify.cash/files/5754745/Expensify.cash.svg.zip)

**Comment by roryabraham** [comment]
Done!

### Web

image

image

### Desktop

image

image

### iOS

image

image

### mWeb

image

**Comment by shawnborton** [comment]
Looks great!

File: ios/Podfile.lock [hunk 1]
```diff
@@ -418,6 +418,8 @@ PODS:
   - urbanairship-react-native (10.0.0):
     - Airship (= 14.2.0)
     - React-Core
+  - RNSVG (12.1.0):
```
Comment (sketchydroide) [inline ios/Podfile.lock:421]: Is this from a library of react?
From Xcode 12 onwards iOS supports svgs, so we souldn't need any pods added to the podfile.

**Review by sketchydroide — CHANGES_REQUESTED** [review]
Looks good for the most parts, just a quick question.
Also there are some merge conflicts.

File: ios/Podfile.lock (hunk 1 above)
Comment (roryabraham) [reply ios/Podfile.lock:421]: Yeah, I believe this is the iOS bridge for `react-native-svg`

**Comment by roryabraham** [comment]
The `RNSVG` pod was not manually added, but was installed as a dependency of:

1) `react-native-svg` - which offers svg support in react-native
2) `react-native-svg-transformer` - which transforms `svg` files into react-native components at compile-time
3) `@svgr/webpack` - which is used by webpack to transform `svg` files into `React` components.

xcode 12 is still relatively new, so I imagine that in future versions `react-native-svg` will eliminate any unnecessary pod dependencies

**Review by sketchydroide — APPROVED** [review]
LGTM

**Comment by sketchydroide** [comment]
> The `RNSVG` pod was not manually added, but was installed as a dependency of:
> 
> 1. `react-native-svg` - which offers svg support in react-native
> 2. `react-native-svg-transformer` - which transforms `svg` files into react-native components at compile-time
> 3. `@svgr/webpack` - which is used by webpack to transform `svg` files into `React` components.
> 
> xcode 12 is still relatively new, so I imagine that in future versions `react-native-svg` will eliminate any unnecessary pod dependencies

That is exactly what I though had happened. But good to confirm.

**Pull request merged**